django/core/management/commands/listurls.py

```diff
@@ -0,0 +1,275 @@
+import json
+from importlib import import_module
+from io import StringIO
+
+from django.conf import settings
+from django.contrib.admindocs.views import (
+    extract_views_from_urlpatterns,
+    simplify_regex,
+)
+from django.core.management import color
+from django.core.management.base import BaseCommand, CommandError, CommandParser
+from django.utils import termcolors
+
+FORMATS = (
+    "aligned",
+    "table",
+    "verbose",
+    "json",
+    "pretty-json",
+)
+
+
+COLORLESS_FORMATS = (
+    "json",
+    "pretty-json",
+)
+
+
+class Command(BaseCommand):
+    help = "List URL patterns in the project with optional filtering by prefixes."
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.style.HEADER = termcolors.make_style(opts=("bold",))
+        self.style.ROUTE = termcolors.make_style(opts=("bold",))
+        self.style.VIEW_NAME = termcolors.make_style(fg="yellow", opts=("bold",))
+        self.style.NAME = termcolors.make_style(fg="red", opts=("bold",))
+
+    def add_arguments(self, parser: CommandParser):
+        super().add_arguments(parser)
+
+        # Sorting
+        parser.add_argument(
+            "--unsorted",
+            "-u",
+            action="store_true",
+            dest="unsorted",
+            help="Filter URLs by the given prefix[es]",
+        )
+
+        # Prefix
+        parser.add_argument(
+            "--prefix",
+            "-p",
+            dest="prefixes",
+            help="Only list URLs with these prefixes.",
+            nargs="*",
+        )
```

Comment:
Just wondering whether `nargs=*` as an option is an anti-pattern or not due to ambiguity? 🤔 

Having a look at all other instances of `nargs=*` in the other commands reveals that they're all regular args. I wonder if this should fall inline with the convention? 🤔 

Though the Python docs mention it's fine to do so 🤷 https://docs.python.org/3/library/argparse.html#nargs

Reply:
I don't think copying the exact same approach from other commands is always a good thing. I needed a way to accept multiple "prefix" arguments and `nargs="*"` does that.